test/integration/controller/jobset_controller_test.go

```diff
@@ -354,6 +378,58 @@ var _ = ginkgo.Describe("JobSet controller", func() {
 				},
 			},
 		}),
+		ginkgo.Entry("suspended jobset will not start", &testCase{
+			makeJobSet: func(ns *corev1.Namespace) *testing.JobSetWrapper {
+				return testJobSet(ns).
+					Suspend(true)
+			},
+			updates: []*jobUpdate{
+				{
+					expectedJobSetCondition: jobset.JobSetSuspended,
+				},
+			},
+		}),
+		ginkgo.Entry("suspended jobset resumes on setting suspend to false", &testCase{
+			makeJobSet: func(ns *corev1.Namespace) *testing.JobSetWrapper {
+				return testJobSet(ns).
+					Suspend(true)
+			},
+			updates: []*jobUpdate{
+				{
+					jobUpdateFn: func(jobList *batchv1.JobList) {
+						suspendJob(&jobList.Items[0])
+						suspendJob(&jobList.Items[1])
+					},
+					expectedJobSetCondition: jobset.JobSetSuspended,
```

Comment:
as mentioned above, add a checkJobState func that iterates over the jobs and checks that they are suspended.

Reply:
Added.